src/main/resources/com/google/api/codegen/ruby/README.md.snip

```diff
@@ -1,47 +1,54 @@
-@snippet generate(metadata)
-  {@metadata.fullName} for Ruby
-  =================================================
-
-  {@metadata.fullName} uses [Google API extensions][google-gax] to provide an
-  easy-to-use client library for the [{@metadata.fullName}][] ({@metadata.majorVersion}) defined in the [googleapis][] git repository
-
-
-  [googleapis]: https://github.com/googleapis/googleapis/tree/master/{@metadata.protoPath}
-  [google-gax]: https://github.com/googleapis/gax-ruby
-  [{@metadata.fullName}]: https://developers.google.com/apis-explorer/?hl=en_US#p/{@metadata.discoveryApiName}/{@metadata.majorVersion}/
-
-  Getting started
-  ---------------
-
-  {@metadata.identifier} will allow you to connect to the [{@metadata.fullName}][] and access all its methods.
-
-  In order to achieve so, you need to set up authentication, as well as install the library locally.
+@extends "ruby/method_sample.snip"
 
+@snippet generate(metadata)
+  {@title(metadata)}
 
-  Setup Authentication
-  --------------------
-
-  To authenticate all of your API calls, first install and setup the [Google Cloud SDK][].
-  Once done, you can then run the following command in your terminal:
+  {@quickStart(metadata)}
 
-      $ gcloud beta auth application-default login
+  {@commonLinkDeclarations(metadata)}
+@end
 
-  or
+@private title(metadata)
+  @# Ruby Client for {@metadata.fullName} ({@metadata.developmentStatus})
 
-      $ gcloud auth login
+  Idiomatic Ruby client for [{@metadata.fullName}][Product Documentation]
 
-  Please see the [gcloud beta auth application-default login][] to find documentation showing the difference between these commands.
+  - [Client Library Documentation][]
+  - [Product Documentation][]
+@end
 
-  [Google Cloud SDK]: https://cloud.google.com/sdk/
-  [gcloud beta auth application-default login]: https://cloud.google.com/sdk/gcloud/reference/beta/auth/application-default/login
+@private quickStart(metadata)
+  @## Quick Start
+  In order to use this library, you first need to go through the following steps:
 
+  - [Select or create a Cloud Platform project.](https://console.cloud.google.com/project)
+  - [Enable the {@metadata.shortName} api.](https://console.cloud.google.com/apis/{@metadata.shortName})
+  - [Setup Authentication.](https://googlecloudplatform.github.io/google-cloud-ruby/@#/docs/google-cloud/master/guides/authentication)
 
-  Installation
-  -------------------
+  @### Installation
+  ```
+  $ gem install {@metadata.identifier}
+  ```
 
-  Install this library using gem:
+  @### Preview
```

Comment:
Here we should put a condition: if the exampleMethods list is empty, we should not show ###Preview section, because sometimes, the smoke test section may not be configured.